cmake/linux/lmms.desktop

```diff
@@ -8,7 +8,7 @@ Comment=Music sequencer and synthesizer
 Comment[ca]=Producció fàcil de música per a tothom!
 Comment[fr]=Production facile de musique pour tout le monde !
 Icon=lmms
-Exec=env QT_X11_NO_NATIVE_MENUBAR=1 QT_AUTO_SCREEN_SCALE_FACTOR=1 lmms %f
```

Comment:
According to [documentation](http://doc.qt.io/qt-5/qt.html), setting `QT_AUTO_SCREEN​_SCALE_FACTOR=1` is enough.

Reply:
Yes but not having to set it is better. And this would make thing not work if lmms was launched differently (ie from the command line).

Which is why the previous change might have been considered working.

Reply:
>Yes but not having to set it is better.

That reason does not convince me. We could drop `QT_X11_NO_NATIVE_MENUBAR=1`, but we do not.

Reply:
Here is *my* rationale:

HiDPI should work out of the box. Nothing to do for the user. `QCoreApplication::setAttribute()` didn't work before, it does now with the patch, make the environment variable `QT_AUTO_SCREEN​_SCALE_FACTOR=1` is not needed anymore. Adding it in the .desktop is only useful when the .desktop is used, which is not ideal.

FWIW, I submitted similar patches to Muse Sequencer and Rosegarden and they got accepted.

Reply:
You should detect whether the screen is HiDPI.

Reply:
Qt does that for you.

Reply:
The Qt documentation says setting `Qt::AA_EnableHighDpiScaling` before constructing `QGuiApplication` is equivalent to setting `QT_AUTO_SCREEN​_SCALE_FACTOR` to 1, as @jasp00 said. However, I agree more with https://github.com/LMMS/lmms/pull/4800#discussion_r252891838. It'd be definitely better to have one which should always work and drop another fix which works only in some situations.

Reply:
It is not a fix. It is an opt-in feature that may incur some scaling or painting artifacts. See "[Migration of Existing Applications](http://doc.qt.io/qt-5/highdpi.html#migration-of-existing-applications)" for the real fix.

Reply:
but you were ready to have it except it wasn't properly activated. And without that you can't start fixing said artifacts.

Reply:
I'm moving this conversation to the conversation area, where it really belongs.